packages/circuits/src/PreventDoubleActionProof.ts

```diff
@@ -0,0 +1,58 @@
+import { Circuit, Prover } from './circuits'
+import { SnarkProof } from '@unirep/utils'
+import { BigNumberish } from '@ethersproject/bignumber'
+import { BaseProof } from './BaseProof'
+
+/**
+ * The prevent double action proof structure that helps to query the public signals
+ */
+export class PreventDoubleActionProof extends BaseProof {
+    readonly idx = {
+        epochKey: 0,
+        stateTreeRoot: 1,
+        control: 2,
+        data: 3,
```

Comment:
I think this is ordered incorrectly, `data` should be the last element because it's a public input.